### ISSUE TITLE
[BUG]There is UNEXPECTED WiFi option in v2307.40000.5.0_x64_Release-Nightly-MindTheGapps-13.0-RemovedAmazon

### ISSUE THREAD

**Issue by Snowlights2022** [MustardChef/WSABuilds#165]
### Describe the bug | 描述一下这个错误

I apologise for some screenshots without English. I added detailed descriptions in English to solve the problem.

<img width="960" alt="屏幕截图 2023-08-26 232925" src="https://github.com/MustardChef/WSABuilds/assets/115212230/09b7b67f-9b69-40a6-b40c-2361de65882d">
<img width="960" alt="屏幕截图 2023-08-26 232940" src="https://github.com/MustardChef/WSABuilds/assets/115212230/0fc6cb6f-3ee4-432f-ad6c-a5c7fdd94a6b">

The screenshot was made when WSA **cannot** connect to Internet.It shows a option ui different from repo's help.There is **no** _VirtWiFi_ and Internet.
I apologise for I didn't make a screenshot when WSA **can** connect to Internet. At that time, it shows _Fake WiFi_ or **nothing**, just like screenshot above.

### Steps to reproduce the issue | 重现该问题的步骤

Advanced setting->Manage developer setting->click search->search WIFI->click WIFI->See the screenshot above

### Expected behaviour | 预期的行为

![261492585-852d8075-69aa-45f1-9ca6-57cff0eefa73](https://github.com/MustardChef/WSABuilds/assets/115212230/5da4350b-daca-4082-9d06-8d5a98e713c8)
Just like screenshot above, and WSA can connect to Internet sucessfully.

### Downloaded Build Of WSA | 已下载的WSA版本

VERSION: v2307.40000.5.0_x64_Release-Nightly-MindTheGapps-13.0-RemovedAmazon
SOURCE: [just from this repo's realease.](https://github.com/MustardChef/WSABuilds/releases/download/Windows_11_2307.40000.5.0/WSA_2307.40000.5.0_x64_Release-Nightly-MindTheGapps-13.0-RemovedAmazon.7z)

### Windows build number | Windows构建号

10.0.22631 Build22631.2262 ni_release

### PC Specification | 个人电脑规格

<img width="648" alt="屏幕截图 2023-08-26 230103" src="https://github.com/MustardChef/WSABuilds/assets/115212230/3308ac52-c99f-4f11-8738-1b59dd3d2eb5">
<img width="791" alt="屏幕截图 2023-08-26 230210" src="https://github.com/MustardChef/WSABuilds/assets/115212230/36893764-ed82-4ec1-a17f-ea1305fe30f2">

### Additional context | 额外的背景

-  #159
   Differently, I saw _FakeWifi_ sometimes when WSA was connected to the Internet. Once I see this again, I will attach the screenshot to this issue.

- Hyper-V settings
  The screenshots were made with Hyper-V Manager form Microsoft.
<img width="539" alt="屏幕截图 2023-08-26 233028" src="https://github.com/MustardChef/WSABuilds/assets/115212230/80c76e40-f943-4553-ad33-6be82dc45774">

   Having tried [the fixes listed in repo's guide](https://github.com/MustardChef/WSABuilds/blob/master/Fixes/FixInternet.md) and using the method _form_ _windows_ _settings,find 'apps'_ _from_ _bar, then find WSA,_ _touch'...',Advanced_ _settings, then_ _try_ _to_ _repair_ _or_ _reset_ , it didn't work. After reset my Windows Internet settings for times, I find that WSA sets virtual network adapter _WSLCore_ for Internet. That is a internal network. But it didn't work, my WSA still has no Internet.
<img width="540" alt="屏幕截图 2023-08-26 233039" src="https://github.com/MustardChef/WSABuilds/assets/115212230/c4d9dd8b-6592-40b8-a289-03f635e68c17">

At the same time, WSL uses virtual network adapter _WSL_ for Internet. That is also internal, but different form WSA, it works. It seems like a bridged network? Once I confirmed it, I will attach details here.
I think there is wrong settings for _WSLCore_ , but I don't know how to find it.  

- Present Situation

   Before I totally reset my Internet settings from Windows settings, WSA can get access to Internet sometimes. It shows _FakeWifi_ in Android settings. At that time it even can cowork with Cloudflare WARP. But after reset, WSL reloaded that _WSL_, and connected to Internet successfully. WSA now has no Internet, **NO Internet**, **NO VPN** can work with WSA. I changed advaced settings _Local_ _network_ _access_ many times, no differences yet.

- If more info needed, just @Sonwlights2022 , I am waiting for solutions!

**Comment by Snowlights2022** [comment]
I want to add that the official WSA 2307.40000.5.0 from ms store works without such problem on my laptop.

**Comment by arishwin** [comment]
here is the screenshot with FakeWifi:
![image](https://github.com/MustardChef/WSABuilds/assets/11484537/1368df6a-6270-42f2-9190-5f12d061610a)

However, I would like to add, that after I click on that, it does not appear in the WiFi settings:
![image](https://github.com/MustardChef/WSABuilds/assets/11484537/b88c388f-65b6-49f7-9d02-a42653f5585e)

And saved networks shows that VirtWifi is disconnected:
![image](https://github.com/MustardChef/WSABuilds/assets/11484537/caf37c5e-dcca-4aad-a648-b03dd5edea68)

**Comment by Snowlights2022** [comment]
There may be bugs from official WSA about this condition. But it seems still unsolved now in v2308. I will stay at v2306 until Microsoft fixed the bug or I get a easier way to alleviate this problem.

**Comment by MustardChef** [comment]
> There may be bugs from official WSA about this condition. But it seems still unsolved now in v2308. I will stay at v2306 until Microsoft fixed the bug or I get a easier way to alleviate this problem.

@Snowlights2022 Does the internet work on the official 2308 builds of WSA?

**Comment by d0j1a1701** [comment]
It seems that this issue is related to the proxy tool (for me, Clash Meta).

When you use a proxy, the system seems to automatically apply the proxy to WSL, this behavior is called "autoproxy", here is the configuration in my `C:/Users/<omit>/.wslconfig`:

```
[experimental]
autoMemoryReclaim=gradual  # gradual | dropcache | disabled
networkingMode=mirrored
dnsTunneling=true
firewall=true
autoProxy=true
```

In this case, WSL will automatically set up an agent consistent with the host, and the Hyper-V Manager shows the WSL virtual switch in "internal" state. Testing has shown that all traffic from WSL goes through the agent.

However, WSA does not seem to automatically set up a proxy. The Hyper-V Manager shows that the WSA virtual switch is in "internal" state, but the WSA settings indicate that there is no Wi-Fi connection and internet access is not available.

So I tried to manually set up a proxy by running the following command in Powershell to manually set WSA to use a host proxy:

```powershell
$WinNetIP=$(Get-NetIPAddress -InterfaceAlias 'vEthernet (Default Switch)' -AddressFamily IPV4)
adb connect 127.0.0.1:58526
adb shell settings put global http_proxy "$($WinNetIP.IPAddress):7890"
```

After the issue was resolved, all traffic from WSA went through the host proxy via the "FakeWifi" network.

**Comment by d0j1a1701** [comment]
Also, I hope developers can add an option to automatically execute a script to set up a proxy after WSA starts.

**Comment by MustardChef** [comment]
@d0j1a1701 Thank you, we will consider this  for our WSAUtilities project. So just to confirm, you were able to successfully set up a proxy with WSA without any issues to the internet. (Most people attempting to use any kind of proxy lose internet access when it is applied).

**Comment by d0j1a1701** [comment]
@MustardChef Yes. The proxy client Clash Meta I use on Windows exposes a mixed type proxy of http/socks5 on 0.0.0.0:7890, all my configurations above are based on this.

**Comment by d0j1a1701** [comment]
@MustardChef I don't know if it's possible to make WSA use the same "mirror" network mode as WSL

**Comment by d0j1a1701** [comment]
Update: I used the "Auto-run script" feature of KernalSU and placed the executable `autoproxy.sh` in `/data/adb/boot-completed.d`.

```
settings put global http_proxy "http://127.0.0.1:7890"
```

Solved the problem.

**Comment by MustardChef** [comment]
> @MustardChef Yes. The proxy client Clash Meta I use on Windows exposes a mixed type proxy of http/socks5 on 0.0.0.0:7890, all my configurations above are based on this.

Does the proxy persist after restart?

**Comment by d0j1a1701** [comment]
> > @MustardChef Yes. The proxy client Clash Meta I use on Windows exposes a mixed type proxy of http/socks5 on 0.0.0.0:7890, all my configurations above are based on this.
> 
> Does the proxy persist after restart?

No, that's why I have to use KernalSU to execute a script everytime startup, in order to automatically set the proxy.

**Comment by d0j1a1701** [comment]
In addition, you should enable "Local Network Access" in the WSA advanced settings and set/add a switch named "vSwitch (WSL)" in the Hyper-V Manager and set it to the network card you are using, so that you can access the proxy on the host using 127.0.0.1.

**Comment by rahaaatul** [comment]
> In addition, you should enable "Local Network Access" in the WSA advanced settings and set/add a switch named "vSwitch (WSL)" in the Hyper-V Manager and set it to the network card you are using, so that you can access the proxy on the host using 127.0.0.1.

How do I add a switch in Hyper-V Manager?

I don't see FakeWifi or VirtWifi in WSA.
